djangodoo/__init__.py

```diff
@@ -100,14 +107,26 @@ def add_extra_model_fields(sender, **kwargs):
 
         The fields are "translated" by using the definitions in fields
     """
+    odoo = getattr(settings, "odoo", None)
+
     def add_field(django_model, field_details):
         odoo_field = convert_field(field_details)
         if odoo_field:
             field = odoo_field.to_django()
             field.contribute_to_class(django_model, field_details['name'])
 
-    odoo = settings.odoo
     if getattr(sender, "_odoo_model", False):
+        # This means we are dealing with an OdooModel
+        if not odoo:
+            # This means set_odoo_client() failed to establish the connexion.
+            # There is no point in proceeding any further as this would raise
+            # an exception that will block the entire application.
+            #TODO handle the case properly instead of bypassing the model generation
+            host = getattr(settings, "ODOO_HOST", False).get("HOST", None)
```

Comment:
For the record, this should never fail as a check is performed [earlier in `set_odoo_client`](https://github.com/fdegrave/djangodoo/pull/4/commits/f2cfa4d7959ce8cc863579a0e4793686d98361a2#diff-44ba29fcf189381fb86d61965b7c7650R26).